src/parquet/column/properties.h

```diff
@@ -126,26 +133,87 @@ class PARQUET_EXPORT WriterProperties {
       return this;
     }
 
-    Builder* encoding(
-        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
-      return encoding(path->ToDotString(), encoding_type);
+    Builder* encodings(const std::shared_ptr<schema::ColumnPath>& path,
+        std::vector<Encoding::type>& encoding_type) {
+      return encodings(path->ToDotString(), encoding_type);
     }
 
-    Builder* encoding(const std::string& column_path, Encoding::type encoding_type) {
+    Builder* encodings(
+        const std::string& column_path, std::vector<Encoding::type>& encoding_type) {
       encodings_[column_path] = encoding_type;
       return this;
     }
 
-    Builder* encoding(Encoding::type encoding_type) {
+    Builder* value_encoding(
+        const std::shared_ptr<schema::ColumnPath>& path, Encoding::type encoding_type) {
+      return value_encoding(path->ToDotString(), encoding_type);
+    }
+
+    Builder* value_encoding(
+        const std::string& column_path, Encoding::type encoding_type) {
+      std::vector<Encoding::type> encodings(3);
+      encodings_[column_path] = encodings;
+      encodings_[column_path][2] = encoding_type;
+      encodings_[column_path][1] = default_dictionary_encoding_;
+      encodings_[column_path][0] = default_level_encoding_;
+      return this;
+    }
```

Comment:
It would be better to have `std::unordered_map<std::string, ColumnEncoding>`, where `ColumnEncoding` is a POD struct than to use `std::vector`.